### DISCUSSION TITLE
Build output Api

### DISCUSSION THREAD

**Discussion by omnaiduu** [vercel/next.js#58271] [App Router]
We anyone could guide how to use Vercel build output api spec 3 to self host nextjs.. or document it would be good... There's no document what parameters does the server less or edge function file in build output take... How to do the routing and use build output api

**Comment by pbzona** [comment]
I think you've misunderstood, the build output API is meant to make frameworks and apps compatible with Vercel deployments, not to self host Next. It's a set of file conventions that determine how Vercel will deploy your app as serverless functions, edge functions, etc.

The signature for serverless functions can vary, but usually will be `request, response`. You can also opt into the native AWS Lambda API, so in that case the signature would be different. You can read more [in this doc](https://vercel.com/docs/functions/serverless-functions/runtimes/node-js).

Edge functions are documented [here](https://vercel.com/docs/functions/edge-functions/edge-functions-api#edge-functions-signature).

Not sure if those docs will be helpful for self hosting, but in case you need to find them now you know where they are 🙂

**Reply by omnaiduu** [reply]
Hey ☺️ those docs where definitely helpful.. thanks.. ig i understand what you said.. but the same build output api is used by cloudflare to self host.. and in coming releases the nextjs is anyways merging the build output api with the Vercel build output api... I just wanted to self host myself using build output.. wanted to know how to go about it.... I want to implement the routing in cloudflare workers but anyone can choose anything... I don't know how to select the appropriate function matching the requests to call to build a response object.. how to use "dest" to locate the function or use entry point.... It would help for other people too

**Comment by mhart** [comment]
Is there an issue tracking Build Output API progress for Next.js? Last word (six months ago) was that the next major version of Next.js was going to output to this – but I don't see much chatter about it in the Next.js 15 announcements.

<img width="516" alt="Screenshot 2024-05-28 at 7 32 13 PM" src="https://github.com/vercel/next.js/assets/367936/6b688be7-29d8-45a2-859f-3b462aa15e5d">

**Reply by mhart** [reply]
[Still says this in the docs](https://nextjs.org/docs/pages/building-your-application/deploying#production-builds) – but couldn't find any issues tracking it specifically.
